uber/site_sections/registration.py

```diff
@@ -318,12 +319,18 @@ def check_in(self, session, id, age_group, group, badge_num=None, message=''):
                     message = 'That badge number already belongs to ' + maybe_dupe.first().full_name
 
             if group:
-                session.match_to_group(attendee, session.group(group))
+                g = sesson.group(group)
+                if g.amount_unpaid:
+                    message = 'That group has an outstanding balance of ${}'.format(g.amount_unpaid)
+                else:
+                    session.match_to_group(attendee, g)
             elif attendee.paid == c.PAID_BY_GROUP and not attendee.group:
                 message = 'You must select a group for this attendee.'
 
-            if attendee.badge_status != c.COMPLETED_STATUS:
-                message = 'This badge is {0} and cannot be checked in.'.format(attendee.badge_status_label)
+            if not message:
+                attendee._status_adjustments()
+                if attendee.badge_status != c.COMPLETED_STATUS:
+                    message = 'This badge is {} and cannot be checked in.'.format(attendee.badge_status_label)
```

Comment:
Two fixes here:

1) If there's already an error message, skip this check so that we show the actual thing blocking the check-in.

2) We need to run ``attendee._status_adjustments()`` before we check their badge status, because otherwise we could be in a situation where a badge which was previously ``c.NEW_STATUS`` should be completed, but errors out before it gets the chance to be marked that way.  This is specifically true for unassigned group badges being claimed.